File: ThematicMapCreator.Api/Contracts/LayerStyleOptions/UniqueValuesStyleOptions.cs
using System;
using System.Collections.Generic;

namespace ThematicMapCreator.Api.Contracts.LayerStyleOptions
{
    public class UniqueValuesStyleOptions : ILayerStyleOptions
    {
        public string PropertyName { get; set; }
        public LayerStyle Style { get; } = LayerStyle.UniqueValues;
        public Dictionary<string, SimpleStyleOptions> ValueStyleOptions { get; set; }
    }
}

File: ThematicMapCreator.Client/src/app/models/layer-style-options/unique-values-style-options.ts
import { LayerStyleOptions } from './layer-style-options';
import { SimpleStyleOptions } from './simple-style-options';
import { LayerStyle } from './layer-style.enum';

export class UniqueValuesStyleOptions implements LayerStyleOptions {
    style = LayerStyle.UniqueValues;
    propertyName: string;
    valueStyleOptions: {[value: string]: SimpleStyleOptions} = {};
}
